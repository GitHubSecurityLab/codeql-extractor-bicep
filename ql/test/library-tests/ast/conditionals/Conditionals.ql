private import bicep

query predicate conditional(Conditionals ast) { any() }

query predicate ifCondition(IfStatement ifStmt, Expr condition, Expr body) {
  ifStmt.getCondition() = condition and
  ifStmt.getBody() = body
}
