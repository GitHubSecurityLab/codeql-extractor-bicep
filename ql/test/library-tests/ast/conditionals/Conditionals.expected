conditional
| conditions.bicep:4:74:12:1 | IfStatement |
ifCondition
| conditions.bicep:4:74:12:1 | IfStatement | conditions.bicep:4:77:4:98 | ParenthesizedExpression | conditions.bicep:4:100:12:1 | Object |
